sycl/unittests/helpers/test/PiMock.cpp

```diff
@@ -0,0 +1,89 @@
+//==--------- PiMock.cpp --- A test for mock helper API's ------------------==//
```

Comment:
We now have tests for tests?

Should we move this file to https://github.com/intel/llvm/tree/sycl/sycl/unittests/pi?